docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+### Pairing Component
+API: [```pairingList.java```](../src/main/java/seedu/duke/PairingList.java)
 
-### Value proposition
+```PairingList``` is responsible for recording which clients renting which property.
 
-{Describe the value proposition: what problem does it solve?}
+```PairingList``` does not inherit from other classes. It stores references to Client and Property objects.
 
-## User Stories
+This a partial class diagram of the ```PairingList``` class:
+
+![Pairing List Design Diagram](diagrams/PairingListDesignDiagram.png)
+```ParsePair``` and ```ParseUnpair``` contain references to data classes ```PairingList```, ```ClientList``` and 
+```PropertyList``` because the data classes provide the required information to validate user input.
+
+### Storage Component
+For `Storage` feature:
+
+![Storage Design Diagram](diagrams/StorageDesignDiagram.jpg)
+
+The Storage class is a superclass itself that is not inherit from other class. This class is responsible for managing 
+three different text file:
+- `client.txt` - Stores the client that is in the Client ArrayList.
+- `property.txt` - Stores the property that is in the Property ArrayList.
+- `pairing.txt` - Stores the relationship between a client and property which is stored in the Pairing hashmap.
+
+
+It has an association with other class which includes:
+- CommandAddProperty
+- CommandAddClient
+- CommandDeleteProperty
+- CommandDeleteClient
+- CommandPair
+- CommandUnpair
+
+Since the arraylist changes by **adding** and **deleting** operations while hashmap changes by **pair** and **unpair** 
+operations, the text files will be updated when `add`, `delete`, `pair` or `unpair` is invoked.
+
+### Common Classes
+___
+## Implementation
+
+This section describes the implementation details of the features within Property Rental Manager.
+
+---
+### Add Feature
+The add feature simply adds an entity to its corresponding list. For Property Rental Manager, there are two variations to the add feature, namely `add -client` and `add -property`.
+
+- `add -client`: Add a new client to the client list.
+- `add -property`: Add a new property to the property list.
+
+The implementation of add feature can be simplified into two major sections. The first section involves the parsing and validation of relevant information provided by the user while the second section comprises the actual addition of client/property to the client list/property list.
+
+<br/>
+
+**Section 1: Parse and Validation of Information**
+
+The first section is facilitated by the following classes:
+
+- `ParseAdd`: Contains common methods used by `ParseAddClient` and `ParseAddProperty`.
+- `ParseAddClient`: Extracts and validates client information from `commandDescription`(User Input).
+- `ParseAddProperty`: Extracts and validates property information from `commandDescription`(User Input).
+
+The following is a simple class diagram of the three classes:
+<p align="center">
+
+![](diagrams/ParseAddRelatedClassesDiagram.jpg)
```

Comment:
this diagram is not shown
<img width="588" alt="Screenshot 2022-10-27 at 5 31 29 PM" src="https://user-images.githubusercontent.com/30253872/198248424-9aff222c-0734-47a6-a302-db0bdf6c4b53.png">